### PR TITLE
chore: add default fsgroup for on podSecurityContext

### PR DESCRIPTION
**Pull request by alfredo-gil**
**Description**
Add a default value for fsgroup on the podSecurityContext that was deleted on this PR [4320](https://github.com/aws/karpenter-provider-aws/pull/4320).

In this PR  [4320](https://github.com/aws/karpenter-provider-aws/pull/4320) there were fixed the security context for the container, but it was deleted the fsgroup on the podSecurityContext level 

**How was this change tested?**
It was deployed with this setup on an eks cluster v1.27 without issues 
**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jonathan-innis** [comment]
We pulled back on some of the security context changes that we made due to some issues that were mentioned in this PR: https://github.com/aws/karpenter-provider-aws/pull/4681. It worries me that adding this change at the pod-level will result in similar problems, since there is no control over the sidecar containers and how to handle the volume mounts.

Perhaps this is something that we allow a sensible default to be set and users that didn't want it would be able to remove it by specifying something like `--set controller.fsGroup=null`?

**Comment by alfredo-gil** [comment]
> We pulled back on some of the security context changes that we made due to some issues that were mentioned in this PR: #4681. It worries me that adding this change at the pod-level will result in similar problems, since there is no control over the sidecar containers and how to handle the volume mounts.
> 
> Perhaps this is something that we allow a sensible default to be set and users that didn't want it would be able to remove it by specifying something like `--set controller.fsGroup=null`?

then my suggestion is to give the option to add a specific `podSecurityContext` on the pod level in case someone need it. with this commit https://github.com/aws/karpenter-provider-aws/pull/5314/commits/ff707ee823cbb5a58766655e62eaafa2dc14bdb6

**Comment by jonathan-innis** [comment]
> Then my suggestion is to give the option to add a specific podSecurityContext on the pod level in case someone need it. with this commit

**Comment by jonathan-innis** [comment]
@alfredo-gil Have any thoughts on this?

**Comment by alfredo-gil** [comment]
@jonathan-innis From my point of view it could be a default value on the fsgroup, for example, on the values.yaml adding:
```yaml
podSecurityContext:
   fsGroup: 65536
```
As you mention this should work for most users and in case any user needs to add specific values or override it to leave it empty they could. I will add the default value on the PR
I don't think that the startup for a sidecar container is going to take longer.

**Comment by jonathan-innis** [comment]
> I don't think that the startup for a sidecar container is going to take longer.

What makes you say this? From what I can read from the docs, this _does_ seem to be the case if there is a large volume: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#:~:text=For%20large%20volumes%2C%20checking%20and%20changing%20ownership%20and%20permissions%20can%20take%20a%20lot%20of%20time

I think, in general, this setting is probably fine since most sidecars won't be mounting sufficiently large volumes, but this should be something that we should callout if we are making this change to our helm chart to default to this setting.

**Comment by jonathan-innis** [comment]
@alfredo-gil I think we are fine taking this in. I'd like to see a callout for this change in the upgrade guide docs if you are willing to take it. Can you make a change to the `website/content/en/preview/upgrading/upgrade-guide.md` path for "Upgrading to v0.34.0+" to callout that this is now configured by default?

**Comment by alfredo-gil** [comment]
@jonathan-innis I'll do. I will add the documentation and I'll let you know.

**Comment by alfredo-gil** [comment]
@jonathan-innis I've just added the documentation in this commit https://github.com/aws/karpenter-provider-aws/pull/5314/commits/8cacadd377e681e6e830c862f4cdb9feea6ab9c0. Do you think that is clear enough? Or should I add something more?

File: website/content/en/preview/upgrading/upgrade-guide.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef
 kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef >}}pkg/apis/crds/karpenter.sh_nodeclaims.yaml
 kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef >}}pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml
 ```
+### Upgrading to v0.34.0+
+
+* Karpenter add now by default on `podSecurityContext` level, specific `fsgroup: 65536`. If you are using sidecar containers, you should review if this configuration is compatible for them. If you need to change this default value you have the option adding specific `podSecurityContext:` block on your values.yaml file
```
Comment (jonathan-innis) [inline website/content/en/preview/upgrading/upgrade-guide.md:37]: There is already a section labeled `Upgrading to v0.34.0` below, can we just add this as a bullet in that section?

Comment (jonathan-innis) [inline website/content/en/preview/upgrading/upgrade-guide.md:37]:
```suggestion
* Karpenter now adds a default `podSecurityContext` that configures the `fsgroup: 65536` of volumes in the pod. If you are using sidecar containers, you should review if this configuration is compatible for them. You can disable this default `podSecurityContext` through helm by performing `--set podSecurityContext={}` when installing/upgrading the chart.
```

A suggestion to add a bit of clarity to the wording

Comment (alfredo-gil) [reply website/content/en/preview/upgrading/upgrade-guide.md]: Changed as suggested with [this commit] (https://github.com/aws/karpenter-provider-aws/pull/5314/commits/b39b5c7b67e2239e6310bf0d17369adbc63e06e2)

Comment (alfredo-gil) [reply website/content/en/preview/upgrading/upgrade-guide.md]: Moved to the proper section with [this commit] (https://github.com/aws/karpenter-provider-aws/pull/5314/commits/b39b5c7b67e2239e6310bf0d17369adbc63e06e2)

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**